src/operation/find_and_modify/mod.rs

```diff
@@ -67,12 +86,20 @@ impl FindAndModify {
             update,
             options.unwrap_or_default(),
         );
-        Ok(FindAndModify { ns, query, options })
+        Ok(FindAndModify {
+            ns,
+            query,
+            options,
+            _phantom: Default::default(),
+        })
     }
 }
 
-impl Operation for FindAndModify {
-    type O = Option<Document>;
+impl<T> Operation for FindAndModify<T>
+where
+    T: Serialize + DeserializeOwned + Unpin + Debug,
+{
+    type O = Option<T>;
```

Comment:
The type `O` for `Operation` requires `Debug`, which necessitates the addition of `Debug` as a requirement for everything upstream. I figured this isn't too onerous for users since they probably already have `Debug` implemented on their types and/or can just derive the trait.

Reply:
Hm yeah, that should probably be okay. Honestly I kind of wish everything derived `Debug` by default in Rust. I forget the reasoning for that not being the case.